File: Maximum_Index.java
class MaxDifference{
    
    // Function to find maximum difference of j-1
    // arr[]: input array
    // n: size of array
    static int maxIndexDiff(int arr[], int n){ 
        
        // Your code here
        
        int ans = 0,end = n-1,front = 0;
        
        while(ans < (n-1-front)){
            end = n-1;
            while(front < end){
                if(arr[front] <= arr[end]){
                    ans = Math.max(ans, end - front);
                    break;
                }
                end--;
            }
            front++;
        }
        return ans;
    }
}

File: Maximum_index.py
for _ in range(int(input())):
    
    n = int(input())
    
    arr = list(map(int, input().split()))
    
    suru,ses = 0,n-1
    
    candidates = []
    
    while suru < n:
    
        while ses >= 0:

            if arr[suru] <= arr[ses]:

                candidates.append(ses - suru)

                break

            ses -= 1

        suru += 1
        
        ses = n-1
        
    print(max(candidates))


